sumatra/recordstore/shelve_store.py

```diff
@@ -38,7 +38,13 @@ class ShelveRecordStore(RecordStore):
 
     def __init__(self, shelf_name=".smt/records"):
         self._shelf_name = shelf_name
+        # Some shelve backends add an extension to the filename, and more than one
```

Comment:
Is this something that is covered by the existing tests? If not, is it worth adding a test for it?